src/mcts/search.cc

```diff
@@ -317,32 +317,51 @@ NNCacheLock Search::GetCachedNNEval(Node* node) const {
 }
 
 void Search::UpdateKLDGain() {
-  if (params_.GetMinimumKLDGainPerNode() <= 0) return;
+  if (params_.GetMinimumKLDGainPerNode() <= 0 &&
+      params_.GetAutoKLDGainMoveFraction() <= 0)
+    return;
 
   SharedMutex::Lock nodes_lock(nodes_mutex_);
   Mutex::Lock lock(counters_mutex_);
-  if (total_playouts_ + initial_visits_ >=
-      prev_dist_visits_total_ + params_.GetKLDGainAverageInterval()) {
+  int interval = params_.GetKLDGainAverageInterval();
```

Comment:
so `KLDGainAverageInterval` is only used if `AutoKLDGainMoveFraction` is not >0, in which case `interval` gets overwritten by `target_playouts * AutoKLDGainIntervalRatio`... so clopping AverageInterval with MoveFraction > 0 would be meaningless?

Reply:
so non-auto mode is used for training, and auto mode is for tc? with no overlap between parameters? we should only be tuning auto parameters for tc then

Reply:
Auto mode is for TC and non-auto for training - that is correct.  Trying to use both will just use auto.